### PR TITLE
feat: add more ruff rules

### PR DESCRIPTION
**Pull request by BeautyyuYanli**

### CODE REVIEW

**Comment by BeautyyuYanli** [comment]
@kemingy mutable data as deafult arg is covered by rule B

File: bindings/python/pyproject.toml [hunk 1]
```diff
@@ -40,19 +40,34 @@ sqlalchemy = [
...
+  "SIM", #https://docs.astral.sh/ruff/rules/#flake8-simplify-sim
+  "N", #https://docs.astral.sh/ruff/rules/#pep8-naming-n
+  "PT", #https://docs.astral.sh/ruff/rules/#flake8-pytest-style-pt
+  "TRY", #https://docs.astral.sh/ruff/rules/#tryceratops-try
+  "FLY", #https://docs.astral.sh/ruff/rules/#flynt-fly
+  "PL", #https://docs.astral.sh/ruff/rules/#pylint-pl
+  "NPY", #https://docs.astral.sh/ruff/rules/#numpy-specific-rules-npy
+  "RUF", #https://docs.astral.sh/ruff/rules/#ruff-specific-rules-ruf
+]
+ignore = ["S101", "E731", "E501"]
```
Comment (kemingy) [inline bindings/python/pyproject.toml:69]: Do we have any `assert` for the runtime code?

File: bindings/python/src/pgvecto_rs/errors.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import numpy as np
+
+
+class VectorError(ValueError):
```
Comment (kemingy) [inline bindings/python/src/pgvecto_rs/errors.py:4]: Might need to make the name more specific to this library name.

File: bindings/python/pyproject.toml (hunk 1 above)
Comment (BeautyyuYanli) [reply bindings/python/pyproject.toml:69]: Not in src but in tests

Comment (kemingy) [reply bindings/python/pyproject.toml:69]: It should be allowed to use `assert` in tests. Seems it's a bug in ruff https://github.com/astral-sh/ruff/issues/4368.

**Review by kemingy — APPROVED** [review]

**Pull request merged**